### DISCUSSION TITLE
Run edge-server on UE physical location type

### DISCUSSION THREAD

**Discussion by alyhasansakr** [InterDigitalInc/AdvantEDGE#185] [Ideas]
**Problem description:**
An automotive application may run on the car itself (physical location), if you use a client server model for this case, then both the client and server are running on the same physical location. Currently the platform does not allow edge applications (servers) to run on physical locations.

**Solution alternatives:**
- Allow edge applications to run on physical locations.
- Allow UE applications to join a service group and to expose a port (just like edge applications).

**Comment by roymx** [comment]
Hello @alyhasansakr,

I'd like to clarify this request.

As per the [model ](https://github.com/InterDigitalInc/AdvantEDGE/raw/master/docs/images/model.png), there are several types of physical locations (UE, Fog, Edge, Cloud)

My understanding is that you would like AdvantEDGE to allow running **Edge-apps** on the **UE Physical Location** type - therefore making the UE a "mobile edge-node"

Please confirm my understanding
Thanks

**Comment by alyhasansakr** [comment]
Yes, that is right I want to run a server (with an exposed port) on a UE physical location.
